core/src/main/java/com/exadel/etoolbox/backpack/core/services/pckg/impl/InstallPackageServiceImpl.java

```diff
@@ -0,0 +1,140 @@
+            packageInfo.setPackageStatus(PackageStatus.INSTALL_IN_PROGRESS);
+            packageInfo.clearLog();
+            packageInfo.addLogMessage(START_INSTALL_MESSAGE + packageInfo.getPackagePath());
+            packageInfo.addLogMessage(LocalDateTime.now().toString());
+            basePackageService.getPackageInfos().put(installPackageModel.getPackagePath(), packageInfo);
+            installPackageAsync(resourceResolver.getUserID(), installPackageModel, packageInfo);
+        }
+        return packageInfo;
+    }
+
+    /**
+     * Called from {@link InstallPackageServiceImpl#installPackage(ResourceResolver, InstallPackageModel)}.
+     * Encapsulates installing package in a separate execution thread
+     *
+     * @param userId User ID per the effective {@code ResourceResolver}
+     * @param installPackageModel {@link InstallPackageModel} object containing user-set options for the package installing
+     * @param packageInfo {@link PackageInfo} object to store package installation status information in
+     */
+    private void installPackageAsync(String userId, InstallPackageModel installPackageModel, PackageInfo packageInfo) {
+        new Thread(() -> installPackage(userId, installPackageModel, packageInfo)).start();
+    }
+
+    /**
+     * Performs the internal package installing procedure and stores status information
+     *
+     * @param userId User ID per the effective {@code ResourceResolver}
+     * @param installPackageModel {@link InstallPackageModel} object containing user-set options for the package installing
+     * @param packageInfo {@link PackageInfo} object to store package installation status information in
+     */
+    private void installPackage(String userId, InstallPackageModel installPackageModel, PackageInfo packageInfo) {
+        Session session = null;
+        try {
+            session = sessionService.getUserImpersonatedSession(userId);
+            JcrPackageManager packMgr = basePackageService.getPackageManager(session);
+            JcrPackage jcrPackage = packMgr.open(session.getNode(packageInfo.getPackagePath()));
+            if (jcrPackage == null) {
+                packageInfo.setPackageStatus(PackageStatus.ERROR);
+                packageInfo.addLogMessage(BasePackageServiceImpl.ERROR + String.format(BasePackageServiceImpl.PACKAGE_DOES_NOT_EXIST_MESSAGE, packageInfo.getPackagePath()));
+                return;
+            }
+            long start = System.currentTimeMillis();
+            ImportOptions importOptions = getImportOptions(installPackageModel, packageInfo);
+            packageInfo.setPackageStatus(PackageStatus.INSTALL_IN_PROGRESS);
+            jcrPackage.install(importOptions);
+            packageInfo.setPackageInstalled(Calendar.getInstance());
+            packageInfo.setPackageStatus(PackageStatus.INSTALL);
+            long finish = System.currentTimeMillis();
+            packageInfo.addLogMessage("Package installed in " + (finish - start) + " ms.");
```

Comment:
I'm afraid, for a package of some 100mb (such packages occur), this will end in "Package installed in 100500436542-hell-only-knows-how-long milliseconds" ))
I'd suggest that you use `org.apache.commons.lang3.time.StopWatch` if it is available in the classpath. 
You add `StopWatch sw = StopWatch.createStarted()` before the long run
and  `sw.toString()` in the end. It converts automatically to a human-readable format like 03:55:551

Reply:
fixed, thanks